NuGetProtocolSamples/Program.cs

```diff
@@ -259,5 +267,71 @@ public static async Task UseNuGetProtocolWithAnAuthenticatedFeed()
             }
             #endregion
         }
+
+        private static async Task PushPackageAsync()
+        {
+            try
+            {
+                // This code region is referenced by the NuGet docs. Please update the docs if you rename the region
+                // or move it to a different file.
+                #region PushPackage
+                ILogger logger = NullLogger.Instance;
+                CancellationToken cancellationToken = CancellationToken.None;
+
+                SourceCacheContext cache = new SourceCacheContext();
+                SourceRepository repository = Repository.Factory.GetCoreV3("https://api.nuget.org/v3/index.json");
+                PackageUpdateResource resource = await repository.GetResourceAsync<PackageUpdateResource>();
+
+                string apiKey = "my-api-key";
+
+                await resource.Push(
+                    "MyPackage.nupkg",
+                    symbolSource: null,
+                    timeoutInSecond: 5 * 60,
+                    disableBuffering: false,
+                    getApiKey: packageSource => apiKey,
+                    getSymbolApiKey: packageSource => null,
+                    noServiceEndpoint: false,
+                    skipDuplicate: false,
+                    symbolPackageUpdateResource: null,
+                    logger);
```

Comment:
This API is funky 😅

Reply:
It's basically mimicking the PushRunner, https://github.com/NuGet/NuGet.Client/blob/5e1970e323f9f55794c6a77562c5535c12d5c2a4/src/NuGet.Core/NuGet.Commands/CommandRunners/PushRunner.cs#L18-L31